test/quantization/core/test_quantized_tensor.py

```diff
@@ -1049,18 +1053,19 @@ def test_qtensor_masked_fill(self):
 
         # the above loop does the same test as test_qtensor_fill
         # now we will check masked_fill for subset of indices
-        mask = torch.randint(0, 2, (numel, ))
+        mask = torch.randint(0, 2, (numel, ), device=device)
         mask = mask.bool()
-        x = torch.rand(numel)
+        x = torch.rand(numel, device=device)
         qx = torch.quantize_per_tensor(x, scale=scale, zero_point=zero_point, dtype=qtype)
         for qtype, fill_with in itertools.product(types, fills):
             q_masked_fill = qx.clone()
             q_masked_fill.masked_fill_(mask, fill_with)
             ref = qx.clone()
+
             for i in range(numel):
                 if mask[i]:
                     # this assignment doesn't end up calling masked_fill, allowing us to compare the different implementations
-                    ref[i] = fill_with
+                    ref[i] = torch.tensor([fill_with], device=device, dtype=torch.float)
```

Comment:
btw you can also just `do torch.tensor(fill_with, device=device, dtype=torch.float)`